### PR TITLE
Get status of VirtualService if available and use that to determine ingress readiness

### LINKED ISSUE #124: Investigate and use Istio 1.6 CRD Status
According to https://istio.io/docs/reference/config/config-status/  Istio CRD Status is now available.

We should investigate and adopt this new feature.

### PR DESCRIPTION
**Pull request by arturenault**
/fixes #124

### CODE REVIEW

**Comment by tcnghia** [comment]
@arturenault it seems istio/istio#28003 is closed. Are we still blocked here?

**Comment by arturenault** [comment]
We were blocked until the release of 1.8.0, which it looks like just happened yesterday. I'll start the process of consuming it.

Once that's in, we can merge this in and start testing it, but won't be able to enable this by default until istio/istio#28797 is fixed.

**Comment by arturenault** [comment]
OK, sorry for the noise. This is finally ready for review @nak3 @ZhiminXiang  :)

**Comment by arturenault** [comment]
@nak3 @JRBANCEL (since zhimin is oof on parental leave), can I get some eyes on this?

File: pkg/reconciler/ingress/ingress.go
Comment (JRBANCEL) [inline pkg/reconciler/ingress/ingress.go:220]: Should this check be first? Probing initialization is not cheap.

File: pkg/reconciler/ingress/ingress.go
Comment (JRBANCEL) [inline pkg/reconciler/ingress/ingress.go:533]: areVirtualServicesReady?

File: pkg/reconciler/ingress/ingress.go
Comment (JRBANCEL) [inline pkg/reconciler/ingress/ingress.go:545]: Should we exit the loop if hasStatus is false?

File: pkg/reconciler/ingress/ingress.go
Comment (JRBANCEL) [inline pkg/reconciler/ingress/ingress.go:587]: Exit loop here?

File: pkg/reconciler/ingress/ingress.go
Comment (arturenault) [reply pkg/reconciler/ingress/ingress.go:220]: Which probing initialization are you talking about? This is called before statusManager.IsReady

Comment (JRBANCEL) [reply pkg/reconciler/ingress/ingress.go:220]: Ah yes, my bad.
`ing.IsReady()` looks a lot like `statusManager.IsReady(ctx, ing)`

**Review by tcnghia — APPROVED** [review]
/lgtm

@arturenault it may be good for us to have some e2e coverage?

may be do this for istio-latest leg?

**Pull request merged**
